### PR TITLE
Python: feat: add chroma memory store

### PR DESCRIPTION
**Pull request by joowon-dm-snu**
### Motivation and Context

#403 

### Description

I've added one quick E2E test example without adding any unit tests. 
I tried to make it as identical to the original classes(`VolatileDataStore` & `VolatileMemoryStore`) as possible.

This PR has 3 issues. 
1. collection name does not allow upper cases([link](https://github.com/chroma-core/chroma/issues/237)). I've written a snake case to replace it inside the current class.
2. `get_nearest_matches_async` function can be more complicated if your team wants to calculate relevance score like [langchain](https://github.com/hwchase17/langchain/blob/master/langchain/vectorstores/chroma.py#L190). 
3. I didn't add requirements (`pip install chromadb`)

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [x] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

**Comment by shawncal** [comment]
FYI: Python PRs on hold while we stage to merge /python into main. See https://github.com/microsoft/semantic-kernel/pull/423

**Comment by shawncal** [comment]
FYI: Python PRs on hold while we stage to merge /python into main. See https://github.com/microsoft/semantic-kernel/pull/423

File: python/semantic_kernel/memory/storage/chroma_data_store.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+        self._client = chromadb.Client(self._client_settings)
+        self._persist_directory = persist_directory
+        self._default_embdding_function = ef.SentenceTransformerEmbeddingFunction()
+        self._default_query_includes = ["embeddings", "metadatas", "documents"]
+
+    async def get_collection_async(self, collection: str) -> Optional["Collection"]:
+        try:
+            # TODO: ChromeDB reject camel case collection names.
+            #   need to decide ChromaDataStore will do auto conversion inside or user takes responseibilities
+            collection_snake_case = camel_to_snake(collection)
```
Comment (awharrison-28) [inline python/semantic_kernel/memory/storage/chroma_data_store.py:60]: I think it's fine to keep the conversion internal to the memory store. This makes it interchangeable, though at small risk of container collision. I'd only really be concerned about this if the collection names were designed to be auto generated.

Comment (joowon-dm-snu) [reply python/semantic_kernel/memory/storage/chroma_data_store.py:60]: If so, I'll leave this as a TODO and we can decide after chroma fixes the issue.

File: python/semantic_kernel/memory/chroma_memory_store.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            (record, distance)
+            for record, distance in zip(
+                self.query_results_to_memory_records(query_results),
+                query_results["distances"][0],
+            )
+        ]
+
+        return output_list
+
+    # TODO: Need to decide semantic-kernel will use chroma's similarity score or not
```
[the remark this answers is not in the document]
Comment (joowon-dm-snu) [reply python/semantic_kernel/memory/chroma_memory_store.py]: I hope this [d418b68](https://github.com/microsoft/semantic-kernel/pull/426/commits/d418b68a209173df4557634b2fc785e0e8de9aae) is what you're talking about

**Comment by dluc** [comment]
@joowon-dm-snu we've merged the python branch into `main`, and GitHub doesn't allow to point this PR to `main` because of some rebase steps (I tried and GH automatically closed the PR, without an option to reopen it). 

Could you send the PR again?

**Pull request closed without merging**